livekit-agents/livekit/agents/pipeline/transcription/synchronizer.py

```diff
@@ -334,42 +333,34 @@ def text_sink(self) -> "_TextSink":
         """Get the text sink wrapper"""
         return self._text_sink
 
-    def _segment_playout_started(self) -> None:
-        self._synchronizer.segment_playout_started()
+    async def _forward_event(self) -> None:
+        while not self._closed:
+            async for segment in self._synchronizer:
+                if not self._base_text_sink:
```

Comment:
under which circumstances would base_text_sink not be defined here? 

would we want to drop the segments in that case or buffer them until the text sink is available?